src/main/scala/io/iohk/ethereum/domain/BlockchainReader.scala

```diff
@@ -71,16 +71,14 @@ class BlockchainReader(
   def getReceiptsByHash(blockhash: ByteString): Option[Seq[Receipt]] = receiptStorage.get(blockhash)
 
   /** get the current best stored branch */
-  def getBestBranch(): Branch =
-    getBestBlock()
-      .map { block =>
-        new BestBranch(
-          block.header,
-          blockNumberMappingStorage,
-          this
-        )
-      }
-      .getOrElse(EmptyBranch$)
+  def getBestBranch(): Branch = {
+    val number = getBestBlockNumber()
+    blockNumberMappingStorage
+      .get(number)
+      .orElse(blockNumberMappingStorage.get(appStateStorage.getBestBlockNumber()))
```

Comment:
What's this for? This is a work-around not related to this change, right?

Reply:
Yes it is not related to this change.
It's related to [anastasiia's comment](https://github.com/input-output-hk/mantis/pull/1073/files/3a61349d1244a5c16b27278508989033a583ea95#r678129983). It allows to be consistent with `getBestBlock` which fallbacks to what is in the storage if the block from memory does not exists.